web/views/relay_group_view.ex

```diff
@@ -1,36 +1,37 @@
 defmodule Cog.V1.RelayGroupView do
   use Cog.Web, :view
 
-  def render("relay_group.json", %{relay_group: relay_group}) do
+  alias Cog.V1.BundlesView
+  alias Cog.V1.RelayView
+
+  def render("relay_group.json", %{relay_group: relay_group}=params) do
     %{id: relay_group.id,
       name: relay_group.name,
-      relays: render_relays(relay_group.relays()),
-      bundles: render_bundles(relay_group.bundles()),
       inserted_at: relay_group.inserted_at,
       updated_at: relay_group.updated_at}
+    |> Map.merge(render_includes(params, relay_group))
   end
+
   def render("index.json", %{relay_groups: relay_groups}) do
-    %{relay_groups: render_many(relay_groups, __MODULE__, "relay_group.json")}
+    %{relay_groups: render_many(relay_groups, __MODULE__, "relay_group.json", include: [:relays, :bundles])}
   end
   def render("show.json", %{relay_group: relay_group}) do
-    %{relay_group: render_one(relay_group, __MODULE__, "relay_group.json")}
+    %{relay_group: render_one(relay_group, __MODULE__, "relay_group.json", include: [:relays, :bundles])}
   end
-  def render("relays.json", %{relay_group: relay_group}) do
-    %{relays: render_relays(relay_group.relays())}
+  def render("members.json", %{relay_group: relay_group}) do
+    %{relays: render_include(:relays, relay_group)}
   end
 
-  defp render_relays(relays) do
-    for relay <- relays do
-      %{id: relay.id,
-        name: relay.name}
-    end
+  defp render_includes(params, relay_group) do
+    Map.get(params, :include, [])
+    |> Enum.reduce(%{}, fn(inc, reply) -> Map.put(reply, inc, render_include(inc, relay_group)) end)
   end
 
-  defp render_bundles(bundles) do
-    for bundle <- bundles do
-      %{id: bundle.id,
-        name: bundle.name}
-    end
+  defp render_include(:bundles, relay_group) do
+    render_many(relay_group.bundles, BundlesView, "bundle.json", as: :bundle)
+  end
+  defp render_include(:relays, relay_group) do
+    render_many(relay_group.relays, RelayView, "relay.json", as: :relay)
```

Comment:
Yeah, it's tough. The fear of creating a loop is exactly the reason for making the `resource.json` version of the template **not** do any includes by default, forcing you to be explicit if you want them. 

My motivation was to centralize the rendered representation of each resource so that when we return a Relay (for example) it's always the same, no matter what URL you're getting it from. It may not include children, but the core resource is the same whether it's `/v1/relays/123` or included under `/v1/relay_groups/456` as a member of a relay group.

I'm not at all sure this is the best approach. Definitely think more about this as we're working on other endpoints and we can see if a better pattern emerges.